test/unittests/statetest_loader_tx_test.cpp

```diff
@@ -0,0 +1,80 @@
+// evmone: Fast Ethereum Virtual Machine implementation
+// Copyright 2023 The evmone Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <gtest/gtest.h>
+#include <test/statetest/statetest.hpp>
+
+using namespace evmone;
+
+TEST(statetest_loader, tx_create_legacy)
+{
+    constexpr std::string_view input = R"({
+        "input": "b0b1",
+        "gas": "9091",
+        "value": "0xe0e1",
+        "sender": "a0a1",
+        "to": "",
+        "gasPrice": "0x7071"
+    })";
+
+    const auto tx = test::from_json<state::Transaction>(json::json::parse(input));
+    EXPECT_EQ(tx.kind, state::Transaction::Kind::legacy);
+    EXPECT_EQ(tx.data, (bytes{0xb0, 0xb1}));
+    EXPECT_EQ(tx.gas_limit, 0x9091);
+    EXPECT_EQ(tx.value, 0xe0e1);
+    EXPECT_EQ(tx.sender, 0xa0a1_address);
+    EXPECT_FALSE(tx.to.has_value());
+    EXPECT_EQ(tx.max_gas_price, 0x7071);
+    EXPECT_EQ(tx.max_priority_gas_price, 0x7071);
```

Comment:
Hm, are these correct? Are they both set to `gasPrice`? I thought the logic is more complex.

Reply:
It turns out you can you EIP-1559 logic this way also for legacy transactions.

Reply:
Yeah, the logic is as simple as that in https://eips.ethereum.org/EIPS/eip-1559:
```python
	def normalize_transaction(self, transaction: Transaction, signer_address: int) -> NormalizedTransaction:
		# legacy transactions
		if isinstance(transaction, TransactionLegacy):
			return NormalizedTransaction(
				signer_address = signer_address,
				signer_nonce = transaction.signer_nonce,
				gas_limit = transaction.gas_limit,
				max_priority_fee_per_gas = transaction.gas_price,
				max_fee_per_gas = transaction.gas_price,
				destination = transaction.destination,
				amount = transaction.amount,
				payload = transaction.payload,
				access_list = [],
			)
```